src/ring_theory/localization/basic.lean

```diff
@@ -435,6 +435,10 @@ lemma ring_hom_ext ⦃j k : S →+* P⦄
   (h : j.comp (algebra_map R S) = k.comp (algebra_map R S)) : j = k :=
 ring_hom.coe_monoid_hom_injective $ monoid_hom_ext M $ monoid_hom.ext $ ring_hom.congr_fun h
 
+lemma alg_hom_subsingleton [algebra R P] : subsingleton (S →ₐ[R] P) :=
```

Comment:
```suggestion
-- TODO[gh-6025]: make this an instance once safe to do so
lemma alg_hom_subsingleton [algebra R P] : subsingleton (S →ₐ[R] P) :=
```

Reply:
this [got a zulip chat](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/subsingleton.20instance.20makes.20a.20structure.20timeout) - why is this related to `simp` if it times out a structure?

Reply:
The subsingleton instance causing timeout probably has to do with `compiler.erase_trivial_structures` calling the `congr` tactic ([see Zulip](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/subsingleton.20instance.20makes.20a.20structure.20timeout/near/293337214)), and the linter isn't happy anyway with this instance having the submonoid as a metavariable. Therefore I only mark it as an instance where I need it.